core/common/src/main/java/alluxio/AlluxioURI.java

```diff
@@ -448,4 +449,18 @@ public String toString() {
     }
     return sb.toString();
   }
+
+  /**
+   * Gets the path component of the {@link AlluxioURI}.
+   *
+   * @return the absolute path
+   */
+  public String getAbsolutePath() {
+    if (mUri.getAuthority() != null) {
+      return PathUtils.concatPath(System.getProperty("user.dir"),
+              mUri.getAuthority().equals(".") ? mUri.getPath() : mUri.getAuthority());
```

Comment:
You'll need to be careful and write some good unit tests to make sure the parsing doesn't miss any edge cases, but it shouldn't be too difficult. I've created a JIRA ticket to track the issue, if you can start working on it please assign it to yourself https://alluxio.atlassian.net/browse/ALLUXIO-2648.

Lets handle it in a separate PR so that it doesn't block the copy-to-directory fix.